test/runtests.jl

```diff
@@ -44,6 +49,19 @@ using ReadOnlyArrays, Test, SparseArrays
                       Base.unsafe_convert(Ptr{Float64}, a)
                 @test stride(r, 1) == stride(a, 1)
             end
+
+            # Test the implicit conversion from mutable array to read-only.
+            f()::ReadOnlyArray{Float64, 2} = a
+            @test f() == a
```

Comment:
you only test here if it does not error. Maybe also check returned value type?

Also - maybe add some test when implicit conversion is invoked (e.g. assigning a value to an array)

Reply:
The implicit conversion happens when returning `a` from inside `f()`, due to its explicit typing of the return value.

Reply:
It shouldn't be possible for the Julia runtime to return a value from `f()` that isn't a `ReadOnlyArray{Float64, 2}`, although if you want I can still add a test for that, for readability.

Reply:
Yes - I know Julia enforces conversion here, but since we want conversion maybe test it.

Reply:
Pushed an update